effekt/shared/src/main/scala/effekt/machine/Analysis.scala

```diff
@@ -41,32 +41,8 @@ def freeVariables(statement: Statement): Set[Variable] =
       freeVariables(rest) - name
     case LiteralDouble(name, value, rest) =>
       freeVariables(rest) - name
+    case LiteralUTF8String(name, utf8, rest) =>
+      freeVariables(rest) - name
     case ForeignCall(name, builtin, arguments, rest) =>
       arguments.toSet ++ freeVariables(rest) - name
   }
-
-def substitute(subst: Substitution, v: Variable): Variable =
-  subst.findLast { case (k, _) => k == v }.map { _._2 }.getOrElse(v)
```

Comment:
Is this not needed anymore? I thought at some point we want to use it?

Reply:
@phischu Said to eliminate this chunk of dead code.